apps/anoma_lib/lib/examples/enock_poly.ex

```diff
@@ -2,7 +2,34 @@ defmodule Examples.ENockPoly do
   use Memoize
 
   require ExUnit.Assertions
+  import ExUnit.Assertions
   import NockPoly
+  alias NockPoly.Term, as: Term
 
   use TypedStruct
+
+  @doc """
+  I test `Term.depth/1` and `Term.size/1` (and thus indirectly also
+  `Term.cata/2`) using a variety of terms.
+  """
+  @spec term_tests() :: bool()
+  def term_tests() do
+    t1 = {1, []}
+    assert Term.depth(t1) == 1
+    assert Term.size(t1) == 1
+
+    t2 = {:a, [{:a, []}, {:b, []}]}
+    assert Term.depth(t2) == 2
+    assert Term.size(t2) == 3
+
+    t3 = {:x, [{:y, [{:z, []}]}]}
+    assert Term.depth(t3) == 3
+    assert Term.size(t3) == 3
+
+    t4 = {0, [{1, []}, {2, [{3, []}, {4, []}]}, {5, []}]}
+    assert Term.depth(t4) == 3
+    assert Term.size(t4) == 6
+
+    true
```

Comment:
I recommend, breaking this into 4 examples...

The main benefit is that you would be able to invoke in your repl the proper term and verify the results yourself!!


```elixir
  def nested_term() do
    res =  {0, [{1, []}, {2, [{3, []}, {4, []}]}, {5, []}]}
    assert Term.depth(res) == 3
    assert Term.size(res) == 6

    res
  end
iex(mariari@Gensokyo)11> Examples.ENockPoly.nested_term
{0, [{1, []}, {2, [{3, []}, {4, []}]}, {5, []}]}
iex(mariari@Gensokyo)12> Examples.ENockPoly.nested_term |> NockPoly.Term.size
6
```

Of course I can then use this to build up more examples

Reply:
Good idea; done in https://github.com/anoma/anoma/pull/1859/commits/d06701a9f9860cce9c1bf86cfcf51a4d9f831f58 .